numba/cuda/tests/cudapy/cache_usecases.py

```diff
@@ -0,0 +1,170 @@
+from numba import cuda
+from numba.cuda.testing import CUDATestCase
+import numpy as np
+import sys
+
+
+class UseCase:
+    """
+    Provide a way to call a kernel as if it were a function.
```

Comment:
Thanks for doing this, it makes it easier to verify and maintain!

Reply:
You're welcome... I did think about trying to align the CUDA and CPU tests even more closely, but I worried it would result in a tangled incomprehensible mess. I think there's no perfect solution, but hopefully this whole setup doesn't cause too much maintenance burden in future :slightly_smiling_face:

Reply:
I think this is a good compromise in terms of alignment vs. complexity, am happy to leave as is!